pkg/synchronizer/synchronizer_test.go

```diff
@@ -268,6 +282,66 @@ func TestSynchronizer(t *testing.T) {
 	assert.EqualValues(t, 2, eventList.Items[0].Count)
 	assert.Equal(t, "new-deploy-id", eventList.Items[0].Annotations[nais_io_v1.DeploymentCorrelationIDAnnotation])
 	assert.Equal(t, nais_io_v1.EventSynchronized, eventList.Items[0].Reason)
+
+	// Ensure that we delete correct IAM-resources from cnrm namespace
+	// Start by creating new app, which also will be the one we delete
+	app2 := fixtures.MinimalApplication()
+	app2.SetAnnotations(map[string]string{
+		nais_io_v1.DeploymentCorrelationIDAnnotation: "deploy-id-2",
+	})
+	app2.ObjectMeta.Name = "iam-test"
+	err = rig.client.Create(ctx, app2)
+	if err != nil {
+		t.Fatalf("Application resource cannot be persisted to fake Kubernetes: %s", err)
+	}
+	// Reconcile the new app
+	req = ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: app2.Namespace,
+			Name:      app2.Name,
+		},
+	}
+	result, err = rig.synchronizer.Reconcile(ctx, req)
+	assert.NoError(t, err)
+	assert.Equal(t, ctrl.Result{}, result)
+
+	// Ensure that we now have two IAMPolicy and two IAMServiceAccount
```

Comment:
In my head, "Ensure" means something like "Make it so". I think "Verify" or even "Assert" would be better, but strictly speaking, do we need this comment at all? Isn't it clear that this is what we are doing?

Reply:
Fjernet alle småkommentarene, men beholder kommentaren som introduserer det jeg tester.